modules/ding_base/ding_base.module

```diff
@@ -535,3 +535,62 @@ function ding_base_rewrite_color_stylesheet() {
   variable_set('color_' . $theme . '_stylesheets', $css);
   variable_set('color_' . $theme . '_files', $paths['files']);
 }
+
+/**
+ * Implements hook_field_widget_form_alter().
+ */
+function ding_base_field_widget_form_alter(&$element, &$form_state, $context) {
+  // Our goal only image fields.
+  if (isset($element['#type']) && $element['#type'] == 'media') {
+    $display = NULL;
+    $displays = $context['instance']['display'];
+    // By default we trying to get default display.
+    if (isset($displays['default']) && $displays['default']['type'] != 'hidden') {
+      $display = $displays['default'];
+    }
+    // Otherwise, we will get first display which has some styles.
+    else {
+      $displays = array_filter($context['instance']['display'], function ($i) {
+        return isset($i['settings']['styles']) && !empty($i['settings']['styles']);
+      });
+      $display = !empty($displays) ? reset($displays) : NULL;
+    }
+
+    if (!empty($display)) {
+      // We will use functionality of image module.
+      $style = image_style_load($display['settings']['styles']['max_style']);
+      // Let's use sample file from image module, for applying all effects.
+      $original_path = variable_get('image_style_preview_image', drupal_get_path('module', 'image') . '/sample.png');
+
+      // Apply style to sample image.
+      $preview_file = image_style_path($style['name'], $original_path);
+      if (!file_exists($preview_file)) {
+        image_style_create_derivative($style, $original_path, $preview_file);
+      }
+
+      // Calculate dimensions.
+      $dimensions = array();
+      $preview_image = image_get_info($preview_file);
+      if ($preview_image['width'] > $preview_image['height']) {
+        $dimensions['width'] = $preview_image['width'];
+        $dimensions['height'] = round($dimensions['width'] / $preview_image['width'] * $preview_image['height']);
+      }
+      else {
+        $dimensions['height'] = $preview_image['height'];
+        $dimensions['width'] = round($dimensions['height'] / $preview_image['height'] * $preview_image['width']);
+      }
+
+      // If after apllying effects dimensions wasn't changed, when we can't recommend.
+      if ($dimensions['height'] != 600 && $dimensions['width'] != 800) {
```

Comment:
These dimensions should be determined by inspecting the sample image. They only work for the default (or one with the exact same dimensions).